test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQOutputBindingTests.cs

```diff
@@ -63,63 +62,35 @@ public void Handles_Null_Attributes_And_Options(string attrHostname, string attr
 
             var loggerFactory = new LoggerFactory();
             var mockServiceFactory = new Mock<IRabbitMQServiceFactory>();
-            var config = new RabbitMQExtensionConfigProvider(new OptionsWrapper<RabbitMQOptions>(opt), mockServiceFactory.Object, (ILoggerFactory)loggerFactory);
+            var mockNameResolver = new Mock<INameResolver>();
+            var config = new RabbitMQExtensionConfigProvider(new OptionsWrapper<RabbitMQOptions>(opt), mockNameResolver.Object, mockServiceFactory.Object, (ILoggerFactory)loggerFactory);
             var actualContext = config.CreateContext(attr);
 
             if (optHostname == null && optQueueName == null)
             {
-                Assert.Equal(actualContext.ResolvedAttribute.Hostname, attrHostname);
+                Assert.Equal(actualContext.ResolvedAttribute.HostName, attrHostname);
                 Assert.Equal(actualContext.ResolvedAttribute.QueueName, attrQueueName);
             }
             else if (attrHostname == null && optQueueName == null)
             {
-                Assert.Equal(actualContext.ResolvedAttribute.Hostname, optHostname);
+                Assert.Equal(actualContext.ResolvedAttribute.HostName, optHostname);
                 Assert.Equal(actualContext.ResolvedAttribute.QueueName, attrQueueName);
             }
             else
             {
-                Assert.Equal(actualContext.ResolvedAttribute.Hostname, optHostname);
+                Assert.Equal(actualContext.ResolvedAttribute.HostName, optHostname);
                 Assert.Equal(actualContext.ResolvedAttribute.QueueName, optQueueName);
             }
         }
 
-        [Fact]
-        public async Task AddAsync_AddsMessagesToQueue()
-        {
-            var mockRabbitMQService = new Mock<IRabbitMQService>(MockBehavior.Strict);
-            var mockBatch = new Mock<IBasicPublishBatch>();
-            mockRabbitMQService.Setup(m => m.GetBatch()).Returns(mockBatch.Object);
-
-            var attribute = new RabbitMQAttribute
-            {
-                Hostname = "localhost",
-                QueueName = "queue",
-            };
-
-            var context = new RabbitMQContext
-            {
-                ResolvedAttribute = attribute,
-                Service = mockRabbitMQService.Object
-            };
-
-            ILoggerFactory loggerFactory = new LoggerFactory();
-            ILogger<RabbitMQAsyncCollector> logger = loggerFactory.CreateLogger<RabbitMQAsyncCollector>();
-            var collector = new RabbitMQAsyncCollector(context, logger);
-
-            byte[] body = Encoding.UTF8.GetBytes("hi");
-            await collector.AddAsync(body);
-
-            mockBatch.Verify(m => m.Add(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>(), It.IsAny<IBasicProperties>(), body), Times.Exactly(1));
-        }
-
         [Fact]
         public void Converts_String_Correctly()

```

Comment:
Move this to its own  test file. Also add a test where conversion fails and throws exception. Add test for null